client/my-sites/customer-home/controller.jsx

```diff
@@ -70,10 +72,14 @@ export async function maybeRedirect( context, next ) {
 			// We need to make sure that sites on the eCommerce plan actually have WooCommerce installed before we redirect to the WooCommerce Home
 			// So we need to trigger a fetch of site plugins
 			await context.store.dispatch( fetchSitePlugins( siteId ) );
+			await context.store.dispatch( fetchModuleList( siteId ) );
```

Comment:
Suggestion: as written, we're going to sequentially call and wait for each call. Might it be better to wrap this code in `Promise.allSettled()` so we can perform both fetches in parallel and wait for both to complete.

This pair of `await` statements may also be impacting the initial rendering of the page, as many parts of the client may continue to run/render rather than waiting for this async code to be executed.

```suggestion
			Promise.allSettled( [
				context.store.dispatch( fetchSitePlugins( siteId ) ),
				context.store.dispatch( fetchModuleList( siteId ) ),
			] ).then(
				///...
			);
```